container-engine-lib/lib/database_accessors/enclave_db/file_artifacts_db/file_artifacts_db.go

```diff
@@ -0,0 +1,166 @@
+	return value, found
+}
+
+func (fileArtifactDb *FileArtifactPersisted) GetArtifactUuidMap() map[string]string {
+	return fileArtifactDb.data.ArtifactNameToArtifactUuid
+}
+
+func (fileArtifactDb *FileArtifactPersisted) DeleteArtifactUuid(artifactName string) {
+	delete(fileArtifactDb.data.ArtifactNameToArtifactUuid, artifactName)
+}
+
+func (fileArtifactDb *FileArtifactPersisted) SetFullUuid(shortenedUuid string, fullUuid []string) {
+	fileArtifactDb.data.ShortenedUuidToFullUuid[shortenedUuid] = fullUuid
+}
+
+func (fileArtifactDb *FileArtifactPersisted) GetFullUuid(shortenedUuid string) ([]string, bool) {
+	value, found := fileArtifactDb.data.ShortenedUuidToFullUuid[shortenedUuid]
+	return value, found
+}
+
+func (fileArtifactDb *FileArtifactPersisted) GetFullUuidMap() map[string][]string {
+	return fileArtifactDb.data.ShortenedUuidToFullUuid
+}
+
+func (fileArtifactDb *FileArtifactPersisted) DeleteFullUuid(shortenedUuid string) {
+	delete(fileArtifactDb.data.ShortenedUuidToFullUuid, shortenedUuid)
+}
+
+func (fileArtifactDb *FileArtifactPersisted) DeleteFullUuidIndex(shortenedUuid string, targetArtifactIdx int) {
+	artifactUuids := fileArtifactDb.data.ShortenedUuidToFullUuid[shortenedUuid]
+	fileArtifactDb.data.ShortenedUuidToFullUuid[shortenedUuid] = append(artifactUuids[0:targetArtifactIdx], artifactUuids[targetArtifactIdx+1:]...)
+}
+
+func (fileArtifactDb *FileArtifactPersisted) SetContentMd5(artifactName string, md5 []byte) {
+	fileArtifactDb.data.ArtifactContentMd5[artifactName] = md5
+}
+
+func (fileArtifactDb *FileArtifactPersisted) GetContentMd5(artifactName string) ([]byte, bool) {
+	value, found := fileArtifactDb.data.ArtifactContentMd5[artifactName]
+	return value, found
+}
+
+func (fileArtifactDb *FileArtifactPersisted) GetContentMd5Map() map[string][]byte {
+	return fileArtifactDb.data.ArtifactContentMd5
+}
+
+func (fileArtifactDb *FileArtifactPersisted) DeleteContentMd5(artifactName string) {
+	delete(fileArtifactDb.data.ArtifactContentMd5, artifactName)
+}
+
+func GetOrCreateNewFileArtifactsDb() (*FileArtifactPersisted, error) {
+	data := fileArtifactData{
+		map[string]string{},
+		map[string][]string{},
+		map[string][]byte{},
+	}
+	db, err := enclave_db.GetOrCreateEnclaveDatabase()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Failed to get enclave database")
+	}
+	fileArtifactPersisted, err := GetFileArtifactsDbFromEnclaveDb(db, &data)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Failed to hydrate pre-existing file artifacts")
+	}
+	return fileArtifactPersisted, nil
+}
+
+func GetFileArtifactsDbFromEnclaveDb(db *enclave_db.EnclaveDB, data *fileArtifactData) (*FileArtifactPersisted, error) {
+	err := db.Update(func(tx *bolt.Tx) error {
+		bucket, err := tx.CreateBucket(takenIpAddressBucketName)
+		if err != nil && err != bolt.ErrBucketExists {
+			return stacktrace.Propagate(err, "An error occurred while creating file artifact bucket")
+		}
+		if err != bolt.ErrBucketExists {
```

Comment:
this condition will be accomplished only if err == nil right? because the previous conditions will be capturing all the errors != nil. if it's true you could simplify the condition just putting `err == nil`

Reply:
Done!